src/components/Button/Button.tsx

```diff
@@ -31,33 +31,34 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
   function Button(
     {
       text,
-      type,
-      size,
+      type = Type.PRIMARY,
+      buttonType,
+      size = Size.MEDIUM,
       leftIcon,
       rightIcon,
       isDisabled,
-      includeHoverAnimations,
+      disabled: disabledProp,
+      includeHoverAnimations = true,
       widthToHideText,
       minWidth,
       className,
       iconClassName,
       textClassName,
-      name,
-      onClick,
-      onMouseEnter,
-      onMouseLeave,
-      onMouseUp,
-      onMouseDown
+      'aria-label': ariaLabelProp,
+      ...other
     },
     ref
   ) {
     const { textIsHidden } = useCollapsibleText(widthToHideText)
+    const disabled = disabledProp ?? isDisabled
+    const isTextVisible = !!text && !textIsHidden
+    const noText = !isTextVisible
 
     const renderLeftIcon = () =>
       leftIcon && (
         <span
           className={cn(iconClassName, styles.icon, styles.left, {
-            [styles.noText]: !text || textIsHidden
+            [styles.noText]: noText
```

Comment:
extra nit-picky: can't this just be `!isTextVisible` to keep from having extra vars?

Reply:
yeah changed!